### PR TITLE
Add typescript typedefs

### PR DESCRIPTION
**Pull request by james-erekson-sportsbet**
This is a simple typedef for react-broadcast. I've exported the prop definitions alongside the components themselves, in case someone needs to mix the props in to one of their related react components.

### CODE REVIEW

**Comment by james-erekson-sportsbet** [comment]
On closer inspection, it looks like this file may not be copied during the build process, do I need to add a line to build.js to copy it over? If so I may just put it in the cjs folder so that the typing is in the same place as the "main" module.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+declare module "react-broadcast" {
+  import { ComponentClass } from "react"
+
+  namespace ReactBroadcast {
```
Comment (j-f1) [inline index.d.ts]: Could you write this as

```ts
declare module "react-broadcast" {
  import { ComponentClass } from "react"

  export interface BroadcastProps {
    // ...
  }
  // etc.
}
```

?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    interface BroadcastProps {
+      channel: string,
+      value?: any
+    }
+
+    interface SubscriberProps {
+      channel: string
```
Comment (j-f1) [inline index.d.ts]: Could you make these generic on the value type and specify the `children` prop as `(value: T) => React.ReactNode`?

**Comment by mjackson** [comment]
I don't personally use TypeScript, so I won't be the best person to maintain these typings as the library changes. I'd recommend you submit them to https://github.com/DefinitelyTyped/DefinitelyTyped instead. Thanks for the PR though 😅

**Pull request closed without merging**